compiler/back_end/util/code_template.py

```diff
@@ -66,6 +66,7 @@ def parse_templates(text):
     A namedtuple object whose attributes are the templates from text.
```

Comment:
Docs should mention that full line comments starting with `//` are removed. It won't matter in most cases but since this uses regex there are weird things like
```
    /*
    some_code();
    // */
```
that would behave differently after stripping.

Reply:
I didn't know you could do that :) (have always assumed line would take precedence for whole line). Perhaps we could update the regex to explicitly not remove that case - line comments that contain a closing block comment?

In addition to updating docs?